### PR TITLE
Batch: pack Batch param to facilitate migration between DPIC/PCIe

### PR DESCRIPTION
**Pull request by klin02**
To facilitate Batch migration between DPIC and PCIe, we pack Batch param to an aligned array, and parse it inside software.

In transmission, we only need to pass single hardware data, as svBitVecVal[] or uint8_t[], and then view it as generated struct in software.

Note we put step inside BatchInfo and call simv_nstep inside, Step will be also exposed to Top all the time for Verilator and Timeout Check.

### CODE REVIEW

File: src/main/scala/Gateway.scala [hunk 1]
```diff
@@ -84,6 +84,7 @@ case class GatewayConfig(
   def check(): Unit = {
     if (hasReplay) require(isSquash)
     if (hasInternalStep) require(isBatch)
+    if (isBatch) require(!hasDutZone)
```
Comment (klin02) [inline src/main/scala/Gateway.scala:87]: In next version of Batch, DiffStates in the same cycle may be splited to different Batch func, so we cannot and no need to pass Batch param to different buffer zones.
To simplify param length division, we disable dutZone with Batch in advance.

File: src/main/scala/DPIC.scala [hunk 2]
```diff
@@ -30,23 +30,28 @@ abstract class DPICBase(config: GatewayConfig) extends ExtModule with HasExtModu
...
   val enable = IO(Input(Bool()))
   val dut_zone = Option.when(config.hasDutZone)(IO(Input(UInt(config.dutZoneWidth.W))))
-  val step = Option.when(config.hasInternalStep)(IO(Input(UInt(config.stepWidth.W))))
 
   def getDirectionString(data: Data): String = {
     if (DataMirror.directionOf(data) == ActualDirection.Input) "input " else "output"
   }
 
-  def getDPICArgString(argName: String, data: Data, isC: Boolean): String = {
+  def getDPICArgString(argName: String, data: Data, isC: Boolean, isDPIC: Boolean = true): String = {
```
Comment (klin02) [inline src/main/scala/DPIC.scala:38]: To support FPGA, which calls v_difftest_Batch inside software and needs `uint8_t io[N]`, we can set isDPIC to false when calling this func to generate dpicFuncProto.

File: src/main/scala/Batch.scala [hunk 3]
```diff
@@ -188,7 +186,10 @@ class BatchEndpoint(bundles: Seq[Valid[DifftestBundle]], config: GatewayConfig,
     }
   }
 
-  val out = IO(Output(new BatchOutput(state_data, state_info, config)))
+  val BatchFinish = WireInit(0.U.asTypeOf(new BatchInfo))
```
Comment (poemonsense) [inline src/main/scala/Batch.scala:189]: No need to use `WireInit`? `Wire(new BatchInfo)` seems fine

**Review by poemonsense — APPROVED** [review]
I don't know all the design details here but the code looks good

File: src/main/scala/Batch.scala (hunk 3 above)
Comment (klin02) [reply src/main/scala/Batch.scala:189]: I will make a change.

**Review by xiaokamikami — COMMENTED** [review]
ok

**Review by poemonsense — APPROVED** [review]

**Pull request merged**